iib/workers/dogpile_cache.py

```diff
@@ -48,7 +49,13 @@ def generate_cache_key(fn, *args, **kwargs):
     arguments = '|'.join(
         [str(arg) for arg in args] + [f'{kwarg}={kwargs[kwarg]}' for kwarg in kwargs]
     )
-    return f'{fn}|{arguments}'
+    key = f'{fn}|{arguments}'
+    if len(key) >= 250:
```

Comment:
why to use hash approach only when key is > 249 characters? Hash should work also with key < 250 right?

Reply:
I was thinking that this was the first time that this happen. And when we hash all the keys we would not be able to "debug" easily - I mean when we would investigate some issue we might want to get data from memcache - and it is easier to read the string created from function name and arguments then SHA number.

But if you think that it is better to count SHA for everything than I can change it.

Reply:
Since others agrees I changed that so SHA256 is always calculated.

Reply:
> And when we hash all the keys we would not be able to "debug" easily

Hmm, have we ever done that actually? And I wonder if we are going to ever do that